src/main/java/org/assertj/core/internal/Strings.java

```diff
@@ -767,6 +773,75 @@ private void assertNotNull(AssertionInfo info, CharSequence actual) {
     Objects.instance().assertNotNull(info, actual);
   }
 
+  /**
+   * Verifies that the given charSequence contains exactly the given charSequences in a sequential order with no more or less elements.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given charSequence.
+   * @param values the charSequences to check.
+   * @throws NullPointerException if the array of values is {@code null}.
+   * @throws IllegalArgumentException if the array of values is empty.
+   * @throws AssertionError if the given {@code CharSequence} is {@code null}.
+   * @throws AssertionError if the given {@code CharSequence} does not contain exactly the values in a sequential order.
+   */
+  public void assertContainsExactly(AssertionInfo info, CharSequence actual, CharSequence[] values) {
+    doCommonCheckForCharSequence(info, actual, values);
+
+    List<String> actualAsList = splitCharSequencesToListOfElements(actual);
+    List<String> valuesAsList = splitCharSequencesToListOfElements(values);
+
+    if (actualAsList.size() != valuesAsList.size()) {
+      throw failures.failure(info, shouldHaveSameSizeAs(actual, actualAsList.size(), valuesAsList.size()));
+    }
+
+    IterableDiff diff = diff(actualAsList, valuesAsList, comparisonStrategy);
+
+    if (!diff.differencesFound()) {
+      // actual and values have the same elements but are they in the same order ?
+      for (int i = 0; i < actualAsList.size(); i++) {
```

Comment:
Maybe it could be cleaner to extract the for loop in a specific method, and it will allow to remove the comment and express it via the method name?

Reply:
agreed

Reply:
Thanks for your advice! I have extracted the code into a new method.😄